### PR TITLE
Adding changes for replacement of key_protect parameter and adding the fix for 3394

### PR DESCRIPTION
**Pull request by IBM-diksha**
Test results :

--- PASS: TestAccIBMCOSKPKmsParamValid (84.76s)
PASS
ok [github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos](https://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos) 86.384s
--- PASS: TestAccIBMCOSKPKmsParamWithInvalidCRN (49.70s)
PASS
ok [github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos](https://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos) 51.390s

--- PASS: TestAccIBMCOSHPCSKmsParamWithInvalidCRN (44.35s)
PASS
ok [github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos](https://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos) 114.080s
--- PASS: TestAccIBMCOSHPCSKmsParamWithInvalidCRN (44.35s)
PASS

ok [github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos](https://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos) (cached)

--- PASS: TestAccIBMCOSKMSBothParamProvided (0.34s)
PASS
ok [github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos](https://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/cos) 1.835s

KMS test cases
=== RUN  TestAccIBMKMSResource_basic
--- PASS: TestAccIBMKMSResource_basic (242.06s)
PASS
ok 	[github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kms](http://github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kms)	244.089s

### CODE REVIEW

**Comment by william8siew** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/tree/master/examples/ibm-kms
You missed this file

**Comment by william8siew** [comment]
And this 
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/examples/ibm-kms/main.tf

**Comment by william8siew** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/examples/ibm-key-protect/README.md

**Comment by william8siew** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/examples/ibm-key-protect/main.tf

**Comment by william8siew** [comment]
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/master/ibm/service/kms/resource_ibm_kms_key_test.go

**Comment by william8siew** [comment]
All the files I have linked above still make reference to 
```
resource "ibm_cos_bucket"
```
using 
**key_protect** and not **kms_key_crn**. Please fix this.
You should also be running the key protect/kms test suites

**Review by william8siew — APPROVED** [review]
lgtm

File: ibm/service/cos/resource_ibm_cos_bucket.go [hunk 1]
```diff
@@ -95,10 +95,18 @@ func ResourceIBMCOSBucket() *schema.Resource {
 				Description: "CRN of resource instance",
 			},
 			"key_protect": {
-				Type:        schema.TypeString,
-				ForceNew:    true,
-				Optional:    true,
-				Description: "CRN of the key you want to use data at rest encryption",
+				Type:          schema.TypeString,
+				ForceNew:      true,
+				Optional:      true,
```
Comment (hkantare) [inline ibm/service/cos/resource_ibm_cos_bucket.go:100]: Do we need to mark this as deprecated

Comment (IBM-diksha) [reply ibm/service/cos/resource_ibm_cos_bucket.go:100]: As of now we are not deprecating the key_protect parameter.
we will just be recommending new customers to use the new parameter.

**Review by hkantare — APPROVED** [review]

**Pull request merged**